src/theory/strings/theory_strings.cpp

```diff
@@ -482,14 +492,15 @@ int TheoryStrings::getReduction( int effort, Node n, Node& nr ) {
           sendInference( d_empty_vec, nnlem, "Reduction", true );
           //we've reduced this n
           Trace("strings-extf-debug") << "  resolve extf : " << n << " based on reduction." << std::endl;
-          return 1;
+          isCd = false;
+          return true;
         }
       }else{
-        return 1;
+        return true;
```

Comment:
Shouldn't we also set `isCd` explicitly here?

Reply:
Yes, although this case should never fire anymore. I'll reorganize/remove this.